pymysql/connections.py

```diff
@@ -1498,19 +1499,24 @@ def _get_descriptions(self):
                     # This behavior is different from TEXT / BLOB.
                     # We should decode result by connection encoding regardless charsetnr.
                     # See https://github.com/PyMySQL/PyMySQL/issues/488
-                    encoding = conn_encoding  # SELECT CAST(... AS JSON) 
-                elif field_type in TEXT_TYPES:
-                    if field.charsetnr == 63:  # binary
-                        # TEXTs with charset=binary means BINARY types.
-                        encoding = None
-                    else:
-                        encoding = conn_encoding
+                    encoding = conn_encoding  # SELECT CAST(... AS JSON)
+                elif field_type in TEXT_TYPES and field.charsetnr != 63:  # binary
+                    encoding = conn_encoding
```

Comment:
This is ugly too.  Since encoding will be overridden, `elif field_type in TEXT_TYPES:` will be enough.
But I believe there must be more simpler code.

Reply:
This is more clean.

```py
if field_type in TEXT_TYPES and field.charsetnr == 63:  # binary
    encoding = converter = None
else:
    converter = self.connection.decoders.get(field_type)
    if field_type in TEXT_TYPES or field_type == FIELD_TYPE.JSON:
        encoding = conn_encoding if use_unicode else None:
    else:
        encoding = 'ascii'
```

Reply:
Oh, wait.  Let's skip converters for text types too.

```py
converter = None
if field_type in TEXT_TYPES and field.charsetnr == 63:  # binary
    encoding = None
elif field_type in TEXT_TYPES or field_type == FIELD_TYPE.JSON:
    encoding = conn_encoding if use_unicode else None
else:
    encoding = 'ascii' if use_unicode else None
    converter = self.connection.decoders.get(field_type)
```